ccl/kvccl/kvtenantccl: TestTenantRangeQPSStat failed

ccl/kvccl/kvtenantccl.TestTenantRangeQPSStat [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8750523?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8750523?buildTab=artifacts#/) on master @ [c95bef097bd4c213c6b5c0c125a9a846c4479d73](https://github.com/cockroachdb/cockroach/commits/c95bef097bd4c213c6b5c0c125a9a846c4479d73):


```
=== RUN   TestTenantRangeQPSStat
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/16471d5fa37cec92efcb07e68ae77ada/logTestTenantRangeQPSStat3808142268
    test_log_scope.go:79: use -show-logs to present logs inline
    tenant_kv_test.go:85: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4406/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/kvccl/kvtenantccl/kvtenantccl_test_/kvtenantccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/kvccl/kvtenantccl/tenant_kv_test.go:85
        	Error:      	"-187" is not greater than "100"
        	Test:       	TestTenantRangeQPSStat
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/16471d5fa37cec92efcb07e68ae77ada/logTestTenantRangeQPSStat3808142268
--- FAIL: TestTenantRangeQPSStat (5.46s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/multi-tenant
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantRangeQPSStat.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-24630

ccl/kvccl/kvtenantccl.TestTenantRangeQPSStat [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8894241?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8894241?buildTab=artifacts#/) on master @ [3036617a2c2f1e9c694857cf02b34e5ce8de5b29](https://github.com/cockroachdb/cockroach/commits/3036617a2c2f1e9c694857cf02b34e5ce8de5b29):


```
=== RUN   TestTenantRangeQPSStat
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/16471d5fa37cec92efcb07e68ae77ada/logTestTenantRangeQPSStat2950694343
    test_log_scope.go:79: use -show-logs to present logs inline
    tenant_kv_test.go:85: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3218/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/kvccl/kvtenantccl/kvtenantccl_test_/kvtenantccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/kvccl/kvtenantccl/tenant_kv_test.go:85
        	Error:      	"-171" is not greater than "100"
        	Test:       	TestTenantRangeQPSStat
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/16471d5fa37cec92efcb07e68ae77ada/logTestTenantRangeQPSStat2950694343
--- FAIL: TestTenantRangeQPSStat (5.79s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantRangeQPSStat.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/kvccl/kvtenantccl.TestTenantRangeQPSStat [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8933152?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8933152?buildTab=artifacts#/) on master @ [14b43be03c1c246765be17aa15245493a42c401d](https://github.com/cockroachdb/cockroach/commits/14b43be03c1c246765be17aa15245493a42c401d):


```
=== RUN   TestTenantRangeQPSStat
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/16471d5fa37cec92efcb07e68ae77ada/logTestTenantRangeQPSStat58190666
    test_log_scope.go:79: use -show-logs to present logs inline
    tenant_kv_test.go:85: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3149/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/kvccl/kvtenantccl/kvtenantccl_test_/kvtenantccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/kvccl/kvtenantccl/tenant_kv_test.go:85
        	Error:      	"-163" is not greater than "100"
        	Test:       	TestTenantRangeQPSStat
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/16471d5fa37cec92efcb07e68ae77ada/logTestTenantRangeQPSStat58190666
--- FAIL: TestTenantRangeQPSStat (5.67s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantRangeQPSStat.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/kvccl/kvtenantccl.TestTenantRangeQPSStat [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8982061?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8982061?buildTab=artifacts#/) on master @ [4dc9e98d8cc1b137ee4cc950a4a42e5895bb890b](https://github.com/cockroachdb/cockroach/commits/4dc9e98d8cc1b137ee4cc950a4a42e5895bb890b):


```
=== RUN   TestTenantRangeQPSStat
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/16471d5fa37cec92efcb07e68ae77ada/logTestTenantRangeQPSStat1254676047
    test_log_scope.go:79: use -show-logs to present logs inline
    tenant_kv_test.go:85: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4431/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/kvccl/kvtenantccl/kvtenantccl_test_/kvtenantccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/kvccl/kvtenantccl/tenant_kv_test.go:85
        	Error:      	"-175" is not greater than "100"
        	Test:       	TestTenantRangeQPSStat
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/16471d5fa37cec92efcb07e68ae77ada/logTestTenantRangeQPSStat1254676047
--- FAIL: TestTenantRangeQPSStat (5.60s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantRangeQPSStat.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/kvccl/kvtenantccl/tenant_kv_test.go
+++ b/pkg/ccl/kvccl/kvtenantccl/tenant_kv_test.go
@@ -31,16 +31,21 @@ func TestTenantRangeQPSStat(t *testing.T) {
 	defer log.Scope(t).Close(t)
 	ctx := context.Background()
 
-	tc := serverutils.StartNewTestCluster(t, 1, base.TestClusterArgs{
-		ServerArgs: base.TestServerArgs{
+	ts, hostDB, _ := serverutils.StartServer(t,
+		base.TestServerArgs{
 			InsecureWebAccess: true,
 			// Must disable test tenant because test below assumes that
 			// it is connecting to the host tenant.
 			DisableDefaultTestTenant: true,
-		},
-	})
-	defer tc.Stopper().Stop(ctx)
-	ts := tc.Server(0)
+			Knobs: base.TestingKnobs{
+				Store: &kvserver.StoreTestingKnobs{
+					// We disable the split queue as an untimely split can cause the QPS
+					// stat to be split over multiple ranges for the tenant.
+					DisableSplitQueue: true,
+				},
+			},
+		})
+	defer ts.Stopper().Stop(ctx)
 
 	_, db := serverutils.StartTenant(
 		t, ts, base.TestTenantArgs{TenantID: serverutils.TestTenantID()},
@@ -54,8 +59,7 @@ func TestTenantRangeQPSStat(t *testing.T) {
 	r.Exec(t, `INSERT INTO foo.qps_test VALUES('abc')`)
 
 	// Host connection.
-	conn := tc.ServerConn(0)
-	sqlDB := sqlutils.MakeSQLRunner(conn)
+	sqlDB := sqlutils.MakeSQLRunner(hostDB)
 
 	var rangeID int
 	stmt := fmt.Sprintf(

